src/main/java/FrameworksDrivers/SignUpView.java

```diff
@@ -1,26 +1,92 @@
 package FrameworksDrivers;
 
-import javax.swing.*;
-import java.awt.event.ActionEvent;
-import java.awt.event.ActionListener;
+import javax.swing.JPanel;
+import java.awt.*;
+import java.awt.event.*;
+import FrameworksDrivers.UIElements.Button;
+import FrameworksDrivers.UIElements.Label;
+import FrameworksDrivers.UIElements.textField;
 
 
 public class SignUpView implements View, ActionListener {
+    Button regB;
 
-    JPanel masterPanel;
+    JPanel masterPanel, newPanel;
+    Label userLabel, passLabel1, passLabel2;
+    textField textFieldUser, textFieldPass1, textFieldPass2;
 
-    public SignUpView(JPanel masterPanel) {
+    CardLayout layout;
+
+    private Object[] paths;
+
+    public SignUpView(JPanel masterPanel, CardLayout layout) {
         //create page to display created element
         this.masterPanel = masterPanel;
+        this.layout = layout;
+
+        this.newPanel = new JPanel();
+        GridLayout newLayout = new GridLayout(4, 2);
+        this.newPanel.setLayout(newLayout);
+        //create components for username input
+        //create label for username input
+        userLabel = new Label();
+        userLabel.createLabel(40 , 40, 40, 40, newPanel, "Username");
+
+        //create text field for inputting username
+        textFieldUser = new textField();
+        textFieldUser.createTextField(newPanel, 40, 40, 40, 40);
+
+
+        //create components for the first password input
+        //create label for password input
+        passLabel1 = new Label();
+        passLabel1.createLabel(40 , 40, 40, 40, newPanel, "Password");
+
+        //create text field for inputting password
+        textFieldPass1 = new textField();
+        textFieldPass1.createTextField(newPanel, 40, 40, 40, 40);
+
+
+        //create components for the second password input
+        // create label for password input
+        passLabel2 = new Label();
+        passLabel2.createLabel(40, 40, 40, 40, newPanel, "Confirm Password");
+
+        //create text field for the second password input
+        textFieldPass2 = new textField();
+        textFieldPass2.createTextField(newPanel, 40, 40, 40, 40);
+        //create buttons for actions
+
```

Comment:
I love all the documentation; sometimes looking at the code for a view model can be a little confusing, this is a big help.